### PR TITLE
FIR-32538: optimisation of FireboltResultSet constructors

### PR DESCRIPTION
**Pull request by alexradzin**

### CODE REVIEW

File: src/main/java/com/firebolt/jdbc/resultset/FireboltResultSet.java [hunk 1]
```diff
@@ -154,25 +115,20 @@ private FireboltResultSet(InputStream is, String tableName, String dbName, Integ
...
 		InputStreamReader inputStreamReader;
 		if (isCompressed) {
 			inputStreamReader = new InputStreamReader(new LZ4InputStream(is), UTF_8);
 		} else {
 			inputStreamReader = new InputStreamReader(is, UTF_8);
 		}
-		return bufferSize != null && bufferSize != 0 ? new BufferedReader(inputStreamReader, bufferSize)
-				: new BufferedReader(inputStreamReader);
+		return new BufferedReader(inputStreamReader, bufferSize);
 	}
```
Comment (ptiurin) [inline src/main/java/com/firebolt/jdbc/resultset/FireboltResultSet.java:132]: Reading the BufferedReader documentation, it throws an exception when buffer size is 0. We used to account for this by setting it with no buffer size (using the default) but now we'll get an exception in this case. 
Should we be throwing it?

**Review by ptiurin — COMMENTED** [review]
Apart from this looks good.

File: src/main/java/com/firebolt/jdbc/resultset/FireboltResultSet.java (hunk 1 above)
Comment (ptiurin) [reply src/main/java/com/firebolt/jdbc/resultset/FireboltResultSet.java:132]: I'll need a bit more explanation. Does setting it in FireboltProperties guarantee we'll never receive 0 here?

Comment (alexradzin) [reply src/main/java/com/firebolt/jdbc/resultset/FireboltResultSet.java:132]: IMHO yes. Moreover we will never get null there. Take a look on the constructor `BufferedReader(Reader)`. It calls `BufferedReader(Reader, defaultBufferSize)` while `defaultBufferSize=8K`. So,  the only thing I did is the code simplification. We always provide reasonable `bufferSize`.

Comment (ptiurin) [reply src/main/java/com/firebolt/jdbc/resultset/FireboltResultSet.java:132]: Discussed offline, should never happen. In any case buffer size 0 should fail.

**Review by ptiurin — APPROVED** [review]

**Pull request merged**